### PR TITLE
feat(be): implement check-duplicated-username api

### LINKED ISSUE #1097: 중복된 username이 존재하는지 확인
### Describe the problem and solution

회원가입 절차에서 중복된 username이 존재하는지 확인하는 API를 구현합니다.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/skkuding/next/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/skkuding/next/blob/main/CONTRIBUTING.md)
- [X] Check that there isn't already an issue

### PR DESCRIPTION
**Pull request by Jaehyeon1020**
### Description

Close #1097 

중복된 username이 존재하는지 확인하는 api를 구현합니다.

### Additional context

---

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/skkuding/next/blob/main/CONTRIBUTING.md)
- [x] Read the [Contributing Guidelines](https://github.com/skkuding/next/blob/main/CONTRIBUTING.md#pr-and-branch) and follow the [Commit Convention](https://github.com/skkuding/next/blob/main/CONTRIBUTING.md#commit-convention)
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: backend/apps/client/src/user/user.controller.ts [hunk 1]
```diff
@@ -180,6 +181,20 @@ export class UserController {
       throw new InternalServerErrorException()
     }
   }
+
+  @Post('username-check')
```
Comment (dotoleeoak) [inline backend/apps/client/src/user/user.controller.ts:185]: 사소하긴 한데, 상태가 변하는 API는 아니니까 GET method로 하고, username은 query로 받는 거 어떤가요?
(`GET /user/username-check?username=foo`)

Comment (Jaehyeon1020) [reply backend/apps/client/src/user/user.controller.ts]: 반영했습니다~!

File: collection/client/User/Check Duplicated Username/Succeed.bru [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+query {
+  username: newusername
+}
+
+body:json {
+  {
+    "username": "newUser"
+  }
+}
```
Comment (dotoleeoak) [inline collection/client/User/Check Duplicated Username/Succeed.bru:21]: body 지워주세요!

File: collection/client/User/Check Duplicated Username/409: Duplicated Username.bru [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+query {
+  username: admin
+}
+
+body:json {
+  {
+    "username": "admin"
+  }
+}
```
Comment (dotoleeoak) [inline collection/client/User/Check Duplicated Username/409: Duplicated Username.bru:21]: 이 부분도 지워주세욥

File: collection/client/User/Check Duplicated Username/Succeed.bru (hunk 2 above)
Comment (Jaehyeon1020) [reply collection/client/User/Check Duplicated Username/Succeed.bru:21]: 헉 오키 알겠습니다

File: collection/client/User/Check Duplicated Username/409: Duplicated Username.bru (hunk 3 above)
Comment (Jaehyeon1020) [reply collection/client/User/Check Duplicated Username/409: Duplicated Username.bru:21]: 둘 다 반영했습니다~

**Review by dotoleeoak — APPROVED** [review]
좋아여

**Pull request merged**